docs/articles/dockerfile_best-practices.md

```diff
@@ -106,10 +106,12 @@ of the child images is sufficient.  However, certain instructions require
 a little more examination and explanation.
 
 * In the case of the `ADD` and `COPY` instructions, the contents of the file(s)
```

Comment:
Thanks for doing this @charleswhchan.  These sentence can be empowered by a little tweaking.

--> https://gist.github.com/moxiegirl/f8b7c0ac1f98232c3baf

* For the `ADD` and `COPY` instructions, the contents of the file(s)
in the image are examined and a checksum is calculated for each file. 
The last-modified and last-accessed times of the file(s) are not considered in these checksums.
During the cache lookup, the checksum is compared against the checksum in the existing images.
If anything has changed in the file(s), such as the contents and metadata,
then the cache is invalidated.